### PR TITLE
[tests-only][full-ci] Add API tests for deleting groups (graph API)

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
Added API tests for deleting groups

Added scenarios:
1. `Scenario Outline: admin user deletes a group`
```feature
      | simplegroup         | nothing special here                  |
      | España§àôœ€         | special European and other characters |
      | नेपाली                | Unicode group name                    |
      | brand-new-group     | dash                                  |
      | the.group           | dot                                   |
      | left,right          | comma                                 |
      | 0                   | The "false" group                     |
      | Finance (NP)        | Space and brackets                    |
      | Admin&Finance       | Ampersand                             |
      | admin:Pokhara@Nepal | Colon and @                           |
      | maint+eng           | Plus sign                             |
      | $x<=>[y*z^2]!       | Maths symbols                         |
      | Mgmt\Middle         | Backslash                             |
      | 😁 😂               | emoji                                 |
      | maintenance#123     | Hash sign                             |
      | 50%25=0             | %25 literal looks like an escaped "%" |
      | staff?group         | Question mark                         |
      | Mgmt/Sydney      | Slash (special escaping happens)   |
      | Mgmt//NSW/Sydney | Multiple slash                     |
      | priv/subadmins/1 | Subadmins mentioned not at the end |
      | var/../etc       | using slash-dot-dot                |
```
2. `Scenario Outline: admin user deletes a group having % (as only special char) in its name` (unable to GET)
```feature
      | 50%pass             | Percent sign (special escaping happens) |
      | 50%2Eagle           | %2E literal looks like an escaped "."   |
      | 50%2Fix             | %2F literal looks like an escaped slash |

```
3. `Scenario: normal user tries to delete a group`
4. `Scenario: deleted group should not be listed in the sharees list`
5. `Scenario: user should not see share received via deleted group`

## Related Issue
- Part of https://github.com/owncloud/ocis/issues/4937

## How Has This Been Tested?
- test environment:

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiGraph/deleteGroup.feature [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    And the administrator has given "Alice" the role "Admin" using the settings api
+
+
+  Scenario Outline: admin user deletes a group
+    Given group "<group_id>" has been created
+    When user "Alice" deletes group "<group_id>" using the Graph API
+    And the HTTP status code should be "204"
+    And group "<group_id>" should not exist
+    Examples:
+      | group_id            | comment                               |
```
Comment (SagarGi) [inline tests/acceptance/features/apiGraph/deleteGroup.feature:18]: is this comment for `group-id` name?

**Review by SagarGi — COMMENTED** [review]
LGTM :+1:

File: tests/acceptance/features/apiGraph/deleteGroup.feature (hunk 1 above)
Comment (saw-jan) [reply tests/acceptance/features/apiGraph/deleteGroup.feature:18]: I think it's only used to give more info to the group name being created

**Review by ScharfViktor — COMMENTED** [review]
maybe add check that user cannot share to group if group was deleted?
- user1 shares to group "users"
- admin deletes the group "users"
- user1 cannot find group "users" in `Share with people ` dialog:
Screenshot 2022-11-21 at 11 38 50


and that user has no access to incoming share after deleting group
- user1 is member of the group "users"
- user2 shares resource to group "users"
- admin deletes group "users"
- after deleting group, user1 has no access to incoming shares (here share still exist. Existing bug #903)

File: tests/acceptance/features/apiGraph/deleteGroup.feature [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    Given group "<group_id>" has been created
+    When user "Alice" deletes group "<group_id>" using the Graph API
+    And the HTTP status code should be "204"
+    And group "<group_id>" should not exist
+    Examples:
+      | group_id         | comment                            |
+      | Mgmt/Sydney      | Slash (special escaping happens)   |
+      | Mgmt//NSW/Sydney | Multiple slash                     |
+      | priv/subadmins/1 | Subadmins mentioned not at the end |
+      | var/../etc       | using slash-dot-dot                |
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiGraph/deleteGroup.feature:60]: can this be merged with the first scenario? all of the steps seems the same.
maybe examples can be separated by some comments.

File: tests/acceptance/features/apiGraph/deleteGroup.feature [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      | admin:Pokhara@Nepal | Colon and @                           |
+      | maint+eng           | Plus sign                             |
+      | $x<=>[y*z^2]!       | Maths symbols                         |
+      | Mgmt\Middle         | Backslash                             |
+      | 😁 😂               | emoji                                 |
+      | maintenance#123     | Hash sign                             |
+      | 50%25=0             | %25 literal looks like an escaped "%" |
+      | staff?group         | Question mark                         |
+
+  @cannot-create-group
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/acceptance/features/apiGraph/deleteGroup.feature:37]: It should be gone on next push

File: tests/acceptance/features/apiGraph/deleteGroup.feature [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  Scenario Outline: admin user deletes a group
+    Given group "<group_id>" has been created
+    When user "Alice" deletes group "<group_id>" using the Graph API
+    And the HTTP status code should be "204"
+    And group "<group_id>" should not exist
+    Examples:
+      | group_id            | comment                                 |
+      | 50%pass             | Percent sign (special escaping happens) |
+      | 50%2Eagle           | %2E literal looks like an escaped "."   |
+      | 50%2Fix             | %2F literal looks like an escaped slash |
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/acceptance/features/apiGraph/deleteGroup.feature:47]: Nope, not yet.
These examples are expected to fail.

File: tests/acceptance/features/apiGraph/deleteGroup.feature (hunk 2 above)
Comment (saw-jan) [reply tests/acceptance/features/apiGraph/deleteGroup.feature:60]: Yeah, those are passing examples. Can be merged.

File: tests/acceptance/features/apiGraph/deleteGroup.feature [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      | maintenance#123     | Hash sign                             |
+      | 50%25=0             | %25 literal looks like an escaped "%" |
+      | staff?group         | Question mark                         |
+      | Mgmt/Sydney         | Slash (special escaping happens)      |
+      | Mgmt//NSW/Sydney    | Multiple slash                        |
+      | priv/subadmins/1    | Subadmins mentioned not at the end    |
+      | var/../etc          | using slash-dot-dot                   |
+
+  @issue-5083
+  Scenario Outline: admin user deletes a group
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/acceptance/features/apiGraph/deleteGroup.feature:42]: Fair point. But would it be nice to put all the examples under the same issue tag even though they are not related to the issue?
I was just trying to separate passing and failing scenarios (failing scenarios will have issue tagged but not the passing ones)
I guess I must give a better scenario description though.

[the remark this answers is not in the document]
Comment (SwikritiT) [reply tests/acceptance/features/apiGraph/deleteGroup.feature]: yes better scenario description would be nice

**Review by SwikritiT — COMMENTED** [review]
LGTM

File: tests/acceptance/features/apiGraph/deleteGroup.feature [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+
+  Scenario: user should not see share received via deleted group
+    Given user "Alice" has uploaded file with content "sample text" to "lorem.txt"
+    And user "Brian" has been created with default attributes and without skeleton files
+    And group "grp1" has been created
+    And user "Brian" has been added to group "grp1"
+    And user "Alice" has shared file "lorem.txt" with group "grp1"
+    And user "Brian" has accepted share "/lorem.txt" offered by user "Alice"
+    And group "grp1" has been deleted
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiGraph/deleteGroup.feature:82]: Shouldn't this be a When step?

File: tests/acceptance/features/apiGraph/deleteGroup.feature [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+  Background:
+    Given user "Alice" has been created with default attributes and without skeleton files
+    And the administrator has given "Alice" the role "Admin" using the settings api
+
+
+  Scenario Outline: admin user deletes a group
+    Given group "<group_id>" has been created
+    When user "Alice" deletes group "<group_id>" using the Graph API
+    And the HTTP status code should be "204"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiGraph/deleteGroup.feature:15]:
```suggestion
    Then the HTTP status code should be "204"
```

File: tests/acceptance/features/apiGraph/deleteGroup.feature [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+      | Mgmt/Sydney         | Slash (special escaping happens)      |
+      | Mgmt//NSW/Sydney    | Multiple slash                        |
+      | priv/subadmins/1    | Subadmins mentioned not at the end    |
+      | var/../etc          | using slash-dot-dot                   |
+
+  @issue-5083
+  Scenario Outline: admin user deletes a group having % (as only special char) in its name
+    Given group "<group_id>" has been created
+    When user "Alice" deletes group "<group_id>" using the Graph API
+    And the HTTP status code should be "204"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiGraph/deleteGroup.feature:45]:
```suggestion
    Then the HTTP status code should be "204"
```

File: tests/acceptance/features/apiGraph/deleteGroup.feature [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+      | 50%pass             | Percent sign (special escaping happens) |
+      | 50%2Eagle           | %2E literal looks like an escaped "."   |
+      | 50%2Fix             | %2F literal looks like an escaped slash |
+
+
+  Scenario: normal user tries to delete a group
+    Given user "Brian" has been created with default attributes and without skeleton files
+    And group "new-group" has been created
+    When user "Brian" tries to delete group "new-group" using the Graph API
+    And the HTTP status code should be "401"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiGraph/deleteGroup.feature:58]:
```suggestion
    Then the HTTP status code should be "401"
```

File: tests/acceptance/features/apiGraph/deleteGroup.feature [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+  Scenario: user should not see share received via deleted group
+    Given user "Alice" has uploaded file with content "sample text" to "lorem.txt"
+    And user "Brian" has been created with default attributes and without skeleton files
+    And group "grp1" has been created
+    And user "Brian" has been added to group "grp1"
+    And user "Alice" has shared file "lorem.txt" with group "grp1"
+    And user "Brian" has accepted share "/lorem.txt" offered by user "Alice"
+    When user "Alice" deletes group "grp1" using the Graph API
+    Then user "Brian" should not have any received shares
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiGraph/deleteGroup.feature:83]: check for the status codes too

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by ScharfViktor — APPROVED** [review]
Nice. thank you)

**Pull request merged**